python/lsst/sims/catUtils/mixins/sncat.py

```diff
@@ -0,0 +1,298 @@
+import numpy as np
+
+from lsst.sims.catalogs.measures.instance import InstanceCatalog
+from lsst.sims.catalogs.measures.instance import compound
+
+from lsst.sims.photUtils import BandpassDict
+from lsst.sims.catUtils.mixins import CosmologyMixin
+from lsst.sims.catUtils.mixins import PhotometryBase
+import lsst.sims.photUtils.PhotometricParameters as PhotometricParameters
+
+import astropy
+
+from .snObject import SNObject
+from .snUniversalRules import SNUniverse
+
+
+wavelenstep = 0.1
+cosmo = CosmologyMixin()
+
+
+class SNIaCatalog (InstanceCatalog, CosmologyMixin, SNUniverse):
+
+    """
+    `lsst.sims.catalogs.measures.instance.InstanceCatalog` class with SN
+    characterized by the  following attributes
+
+    Attributes
+    ----------
+    position : 3-tuple of floats
+              (ra, dec, redshift),
+    velocity : 3 tuple of floats
+              velocity wrt host galaxy in Km/s,
+    the supernova model (eg. SALT2)
+    and parameters of the supernova model that predict the SED.
+    """
+
+    # t_0, c, x_1, x_0 are parameters characterizing a SALT
+    # based SN model as defined in sncosmo
+    column_outputs = ['snid', 'snra', 'sndec', 'z', 't0', 'c', 'x1',
+                      'x0']
+
+    # You can add parameters like fluxes and magnitudes by adding the following
+    # variables to the list
+    # 'flux_u', 'flux_g', 'flux_r', 'flux_i', 'flux_z', 'flux_y' ,
+    # 'mag_u', 'mag_g', 'mag_r', 'mag_i', 'mag_z', 'mag_y']
+
+    # Flux variables are convenient to display in exponential format to avoid
+    # having them cut off
+    variables = ['flux_u', 'flux_g', 'flux_r', 'flux_i', 'flux_z', 'flux_y']
+    variables += ['flux', 'flux_err', 'mag_err']
+
+    override_formats = {'snra': '%8e', 'sndec': '%8e', 'c': '%8e',
+                        'x0': '%8e'}
+    for var in variables:
+        override_formats[var] = '%8e'
+
+    cannot_be_null = ['x0', 'z', 't0']
+
+    @astropy.utils.lazyproperty
+    def mjdobs(self):
```

Comment:
Be aware: I just merged a change to sims_utils.  ObservationMetaData.mjd is now an instance of the class ModifiedJulianDate, which is a thin wrapper of astropy time.  You probably want to return something like self.obs_metadata.mjd.TAI here.